src/Adapter/Product/SpecificPrice/CommandHandler/EditProductSpecificPriceHandler.php

```diff
@@ -0,0 +1,146 @@
+            $specificPrice,
+            $this->fillUpdatableProperties($command, $specificPrice)
+        );
+    }
+
+    /**
+     * @param EditProductSpecificPriceCommand $command
+     * @param SpecificPrice $specificPrice
+     *
+     * @return string[]
+     */
+    private function fillUpdatableProperties(EditProductSpecificPriceCommand $command, SpecificPrice $specificPrice): array
+    {
+        $updatableProperties = [];
+        if (null !== $command->getReduction()) {
+            $specificPrice->reduction_type = $command->getReduction()->getType();
+            $specificPrice->reduction = (float) (string) $command->getReduction()->getValue();
+            $updatableProperties = [
+                'reduction_type',
+                'reduction',
+            ];
+        }
+
+        if (null !== $command->includesTax()) {
+            $specificPrice->reduction_tax = $command->includesTax();
+            $updatableProperties[] = 'reduction_tax';
+        }
+
+        if (null !== $command->getPrice()) {
+            $specificPrice->price = (float) (string) $command->getPrice();
+            $updatableProperties[] = 'price';
+        }
+
+        if (null !== $command->getFromQuantity()) {
+            $specificPrice->from_quantity = $command->getFromQuantity();
+            $updatableProperties[] = 'from_quantity';
+        }
+
+        if (null !== $command->getShopGroupId()) {
+            $specificPrice->id_shop_group = $command->getShopGroupId()->getValue();
+            $updatableProperties[] = 'id_shop_group';
+        }
+
+        if (null !== $command->getShopId()) {
+            $specificPrice->id_shop = $command->getShopId()->getValue();
+            $updatableProperties[] = 'id_shop';
+        }
+
+        if (null !== $command->getCombinationId()) {
+            $specificPrice->id_product_attribute = $command->getCombinationId()->getValue();
+            $updatableProperties[] = 'id_product_attribute';
+        }
+
+        if (null !== $command->getCurrencyId()) {
+            $specificPrice->id_currency = $command->getCurrencyId()->getValue();
+            $updatableProperties[] = 'id_currency';
+        }
+
+        if (null !== $command->getCountryId()) {
+            $specificPrice->id_country = $command->getCountryId()->getValue();
+            $updatableProperties[] = 'id_country';
+        }
+
+        if (null !== $command->getGroupId()) {
+            $specificPrice->id_group = $command->getGroupId()->getValue();
+            $updatableProperties[] = 'id_group';
+        }
+
+        if (null !== $command->getCustomerId()) {
+            $specificPrice->id_customer = $command->getCustomerId()->getValue();
+            $updatableProperties[] = 'id_customer';
+        }
+
+        if (null !== $command->getDateTimeFrom()) {
+            $specificPrice->from = $command->getDateTimeFrom()->format(DateTime::DEFAULT_DATETIME_FORMAT);
+        }
+
+        if (null !== $command->getDateTimeTo()) {
```

Comment:
Also a question regarding from/to do they make sense independently? Since it is an interval do they need to be both defined? Or is it possible to define only the start date or only the end date?

In the validator it seems like the rule is only enforced when both are defined. Does it mean you MUST define both? Or is it possible to set them separately but the interval rule only makes sense when both are defined?

In case both need to be defined there should be a rule in the VO that enforces both fields are set once at least one is set.

Reply:
its possible to leave one, other or both null value (0000-00-00).
So you can define it like "from now till unlimited" or "from unlimited till now" or just "unlimited" (if both are nullvalue)